client/src/components/landing_page/LandingPageSection.vue

```diff
@@ -0,0 +1,100 @@
+<template>
+  <div class='landing-page-section' :style='cssProps'>
+    <div class='header-section'>
+      <h2>{{ messages.SUPER_HEADER }}</h2>
+      <h1>{{ messages.HEADER }}</h1>
+      <h2>{{ $props.messages.BODY }}</h2>
+    </div>
+    <slot></slot>
+  </div>
+</template>
+
+<script lang='ts'>
+import { defineComponent, PropType } from 'vue';
+import { LandingPageStyles, Style } from '../../assets/styles/style_props';
+
+export interface Messages {
+  SUPER_HEADER?: string;
+  HEADER: string;
+  BODY: string;
+  CTA_PLACEHOLDER?: string;
+  CTA_BUTTON?: string;
+}
+
+/**
+ * A component that displays basic content on the landing page. It can accept inner components for
+ * extra functionality, such as buttons, links, or videos. The purpose of this component is to
+ * keep the landing page DRY and define style in one place.
+ */
+export default defineComponent({
+  name: 'LandingPageSection',
+  // TODO: add props to customize basic styling. E.g. background color.
+  props: {
+    messages: {
+      type: Object as PropType<Messages>,
+      required: true,
+    },
+    styles: {
+      type: Object as PropType<Style>,
+      required: true,
+    },
+  },
+  computed: {
+    cssProps(): Object {
+      return {
+        '--header-text-color': this.styles.headerTextColor ?? LandingPageStyles.DEFAULT_STYLE.headerTextColor,
+        '--subheader-text-color': this.styles.subHeaderTextColor ?? LandingPageStyles.DEFAULT_STYLE.subHeaderTextColor,
+        '--header-text-size': this.styles.headerTextSize ?? LandingPageStyles.DEFAULT_STYLE.headerTextSize,
+        '--subheader-text-size': this.styles.subheaderTextSize ?? LandingPageStyles.DEFAULT_STYLE.subheaderTextSize,
+        '--background-color': this.styles.backgroundColor ?? LandingPageStyles.DEFAULT_STYLE.backgroundColor,
+        '--height': this.styles.height ?? LandingPageStyles.DEFAULT_STYLE.height,
+      };
+    },
+  },
+});
+</script>
+
+<style scoped>
+h1, h2 {
```

Comment:
Cautioning against using h1, h2 headers since you have to overwrite so much css. in global styles theres predefined classes for this so that if we decide to change the h1 for mobile or something it will be applied throughout + less overwrites have to happen with padding and such

Reply:
Didn't realize we had classes for this and Brett left a comment on last PR to the opposite effect, so have been using h1/h2 consistently. Agree it makes sense to use one or the other. I'll remove h1/h2.

Reply:
Switched to shared header classes (additionally added large + xl sizes where needed).